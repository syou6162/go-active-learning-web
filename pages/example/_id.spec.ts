import { createLocalVue, shallowMount } from '@vue/test-utils'
import { Auth } from 'aws-amplify';
import { BButton, BCard, BCardBody, BCardFooter, BListGroup, BListGroupItem } from 'bootstrap-vue'
import VueLazyLoad from 'vue-lazyload'
import VueMeta from 'vue-meta'
import ExamplePage from './_id.vue'

const localVue = createLocalVue();
localVue.use(VueLazyLoad);
localVue.component('b-button', BButton)
localVue.component('b-card', BCard)
localVue.component('b-card-body', BCardBody)
localVue.component('b-card-footer', BCardFooter)
localVue.component('b-list-group', BListGroup)
localVue.component('b-list-group-item', BListGroupItem)
// ref: https://stackoverflow.com/questions/59964001/how-to-test-head-in-nuxt-js-using-jest
localVue.use(VueMeta, { keyName: 'head' })

const hatenaBookmark = {
  title: "Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)",
  screenshot: "https://b.st-hatena.com/images/v4/public/common/noimage.png?version=8358883de4e4eced6d5c2b49020f40bbbcfdca71",
  entry_url: "https://b.hatena.ne.jp/entry/s/blog.trocco.io/dev-articles/embulk-digdag-meetup-2020",
  bookmarks: [
    {
      "comment": "",
      "user": "machupicchubeta",
      "tags": [],
      "timestamp": "2020/08/27 00:56"
    },
    {
      "tags": [],
      "timestamp": "2020/08/26 19:53",
      "user": "hoppie",
      "comment": ""
    },
    {
      "comment": "“統合”",
      "user": "polamjag",
      "timestamp": "2020/08/26 14:29",
      "tags": []
    },
    {
      "comment": "",
      "user": "yukiyan_w",
      "tags": [
        "digdag",
        "embulk"
      ],
      "timestamp": "2020/08/26 12:15"
    }
  ],
  count: 5,
  url: "https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020",
  eid: "4690501516113184834"
}

const referringTweets = {
  Tweets: [
    {
      ExampleId: 21805423,

      CreatedAt: "2020-08-26T03:13:06Z",
      IdStr: "1298458422581448704",
      FullText: "【イベント情報】先月開催された「Embulk &amp; DigDag Online Meetup 2020」のレポート記事を公開しました。\nEmbulkとDigdagをプロダクション環境で利用している事例や、Embulk開発者による発表が盛り沢山です！是非ご一読下さい。\nhttps://t.co/gdtRuoyJtY",
      FavoriteCount: 6,
      RetweetCount: 4,
      Lang: "ja",

      ScreenName: "trocco_jp",
      Name: "データ分析基盤の総合支援SaaS「trocco(トロッコ)」",
      ProfileImageUrl: "https://pbs.twimg.com/profile_images/1051692644311556096/WmDcCj4Z_normal.jpg",
      Label: 1,
      Score: 0.4894515842007917
    },
    {
      ExampleId: 21805423,

      CreatedAt: "2020-08-26T03:13:06Z",
      IdStr: "1111111111",
      FullText: "適当な負例です",
      FavoriteCount: 6,
      RetweetCount: 4,
      Lang: "ja",

      ScreenName: "syou6162",
      Name: "データ分析基盤の総合支援SaaS「trocco(トロッコ)」",
      ProfileImageUrl: "https://pbs.twimg.com/profile_images/1051692644311556096/WmDcCj4Z_normal.jpg",
      Label: 0,
      Score: -2.0
    },
  ], 
  Count: 2
}

const example = {
  Id: 11111,
  Label: 0,
  Url: "https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020",
  FinalUrl: "https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020",
  Title: "Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)",
  Description: "",
  HatenaBookmark: hatenaBookmark,
  ReferringTweets: referringTweets,
}

const keywords = ["embulk", "digdag"]

describe('ExamplePage', () => {
  jest.spyOn(Auth, 'currentAuthenticatedUser').mockReturnValue(
    Promise.resolve({
      username: "myuser",
      attributes: {
        email: "test@test.com",
        email_verified: true,
        phone_number: "08012345678"
        // ...other attoributes
      }
      // ...other parameters
    })
  );

  const wrapper = shallowMount(ExamplePage, {
    localVue,
  });

  test('メタタグが正しく出ている', async () => {
    await wrapper.setData({
      title: example.Title,
      example: example,
      keywords: keywords,
    })

    expect(wrapper.exists()).toBeTruthy()
    // ref: https://stackoverflow.com/questions/59964001/how-to-test-head-in-nuxt-js-using-jest
    // @ts-ignore
    expect(wrapper.vm.$metaInfo.meta.find((item) => item.name === 'robots').content).toBe('index, follow')
  })
  
  test('負例は検索エンジンから無視してもらう', async () => {
    await wrapper.setData({
      title: example.Title,
      example: {
        Label: -1
      },
      keywords: keywords,
    })

    expect(wrapper.exists()).toBeTruthy()
    // ref: https://stackoverflow.com/questions/59964001/how-to-test-head-in-nuxt-js-using-jest
    // @ts-ignore
    expect(wrapper.vm.$metaInfo.meta.find((item) => item.name === 'robots').content).toBe('noindex, nofollow')
    // @ts-ignore
    expect(wrapper.vm.$metaInfo.meta.find((item) => item.name === 'description').content).toBeDefined()
    // @ts-ignore
    expect(wrapper.vm.$metaInfo.meta.find((item) => item.name === 'og:title').content).toBeDefined()
  })

  test('Twitterの項目が出ている', async () => {
    await wrapper.setData({
      title: example.Title,
      example: example,
      keywords: keywords,
    })

    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.find('div#referring-tweet').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })
  
  test('はてなブックマークの項目が出ている', async () => {
    await wrapper.setData({
      title: example.Title,
      example: example,
      keywords: keywords,
    })

    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.find('div#bookmark-comments').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('ExamplePage#Amplify', () => {
  test('amplifyを通っていないときはアノテーションボタンを出さない', async () => {
    jest.spyOn(Auth, 'currentAuthenticatedUser').mockReturnValue(
      Promise.reject(new Error('Not Authenticated User'))
    )
    const wrapper = shallowMount(ExamplePage, {
      localVue,
    });
    await wrapper.setData({
      title: example.Title,
      example: example,
      keywords: keywords,
    })
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.find('annotate-buttons-stub').exists()).toBe(false)
    expect(wrapper.find('tweet-annotate-buttons-stub').exists()).toBe(false)
    expect(wrapper.element).toMatchSnapshot()

    // 無理やり書き換えると表示できることを確認
    await wrapper.setData({
      title: example.Title,
      example: example,
      keywords: keywords,
      isAdmin: true
    })
    expect(wrapper.find('annotate-buttons-stub').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })
  test('amplifyを通っているときはアノテーションボタンを出す', async () => {
    jest.spyOn(Auth, 'currentAuthenticatedUser').mockReturnValue(
      Promise.resolve({
        username: "myuser",
        attributes: {
          email: "test@test.com",
          email_verified: true,
          phone_number: "08012345678"
          // ...other attoributes
        }
        // ...other parameters
      })
    );

    const wrapper = shallowMount(ExamplePage, {
      localVue,
    });
    await wrapper.setData({
      title: example.Title,
      example: example,
      keywords: keywords,
    })
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.find('annotate-buttons-stub').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })
})
