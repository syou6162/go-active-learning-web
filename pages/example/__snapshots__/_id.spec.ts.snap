// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExamplePage Twitterの項目が出ている 1`] = `
<div
  class="mx-auto example"
>
  <b-card-stub
    bodytag="div"
    footertag="div"
    headertag="div"
    subtitletag="h6"
    subtitletextvariant="muted"
    tag="article"
    titletag="h4"
  >
    <h1
      class="h4"
    >
      
      Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
    
    </h1>
     
    <!---->
     
    <p
      class="card-text"
    >
      
      Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
    
    </p>
     
    <div>
      <a
        class="hatena-bookmark-link"
        href="https://b.hatena.ne.jp/entry/s/blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
        rel="noopener"
        target="_blank"
      >
        5 users
      </a>
      :
      
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
    </div>
     
    <div>
      <span
        class="tweets-count"
      >
        2 mentions
      </span>
      :
      
      <twitter-icon-stub
        tweet="[object Object]"
      />
      <twitter-icon-stub
        tweet="[object Object]"
      />
    </div>
     
    <div>
      
      Keywords: 
      
      <b-button-stub
        class="keyword"
        href="/search?query=embulk"
        size="sm"
        tag="button"
        target="_self"
        type="button"
        variant="outline-primary"
      >
        
        embulk
      
      </b-button-stub>
      <b-button-stub
        class="keyword"
        href="/search?query=digdag"
        size="sm"
        tag="button"
        target="_self"
        type="button"
        variant="outline-primary"
      >
        
        digdag
      
      </b-button-stub>
    </div>
    
    Date: 
    
    <annotate-buttons-stub
      example="[object Object]"
    />
     
    <b-card-footer-stub
      footertag="div"
    >
      <!---->
       
      <a
        href="https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
        rel="noopener"
        target="_blank"
      >
        blog.trocco.io 
      </a>
       
      <b-button-stub
        class="float-right"
        href="https://twitter.com/intent/tweet?text=👀&hashtags=ml_news&url=https://www.machine-learning.news/example/11111"
        rel="noopener"
        size="sm"
        tag="button"
        target="_blank"
        type="button"
        variant="primary"
      >
        
        Tweet
      
      </b-button-stub>
    </b-card-footer-stub>
  </b-card-stub>
   
  <div
    id="referring-tweet"
  >
    <h2
      class="h4"
    >
      
      Referring Tweets
    
    </h2>
     
    <b-list-group-stub
      tag="div"
    >
      <referring-tweet-stub
        isadmin="true"
        tweet="[object Object]"
      />
    </b-list-group-stub>
  </div>
   
  <div
    id="bookmark-comments"
  >
    <h2
      class="h4"
    >
      
      Bookmark Comments
    
    </h2>
     
    <b-list-group-stub
      tag="div"
    >
      <b-list-group-item-stub
        class="hatena-bookmark-item"
        tag="div"
        target="_self"
      >
        <img
          class="hatena-bookmark-comment-user-icon-img"
          data-src="https://cdn.profile-image.st-hatena.com/users/polamjag/profile.png"
          lazy="loading"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
         
        <div
          class="hatena-bookmark-user-link-and-comment-container"
        >
          <a
            class="hatena-bookmark-user-link"
            href="http://b.hatena.ne.jp/polamjag"
            rel="noopener"
            target="_blank"
          >
            
            id:polamjag
          
          </a>
           
          <span
            class="hatena-bookmark-comment"
          >
            
            “統合”
          
          </span>
           
          <span
            class="hatena-bookmark-timestamp"
          >
            
            2020/08/26 14:29
          
          </span>
        </div>
      </b-list-group-item-stub>
    </b-list-group-stub>
  </div>
   
  <!---->
   
  <b-list-group-stub
    tag="div"
  />
</div>
`;

exports[`ExamplePage はてなブックマークの項目が出ている 1`] = `
<div
  class="mx-auto example"
>
  <b-card-stub
    bodytag="div"
    footertag="div"
    headertag="div"
    subtitletag="h6"
    subtitletextvariant="muted"
    tag="article"
    titletag="h4"
  >
    <h1
      class="h4"
    >
      
      Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
    
    </h1>
     
    <!---->
     
    <p
      class="card-text"
    >
      
      Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
    
    </p>
     
    <div>
      <a
        class="hatena-bookmark-link"
        href="https://b.hatena.ne.jp/entry/s/blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
        rel="noopener"
        target="_blank"
      >
        5 users
      </a>
      :
      
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
    </div>
     
    <div>
      <span
        class="tweets-count"
      >
        2 mentions
      </span>
      :
      
      <twitter-icon-stub
        tweet="[object Object]"
      />
      <twitter-icon-stub
        tweet="[object Object]"
      />
    </div>
     
    <div>
      
      Keywords: 
      
      <b-button-stub
        class="keyword"
        href="/search?query=embulk"
        size="sm"
        tag="button"
        target="_self"
        type="button"
        variant="outline-primary"
      >
        
        embulk
      
      </b-button-stub>
      <b-button-stub
        class="keyword"
        href="/search?query=digdag"
        size="sm"
        tag="button"
        target="_self"
        type="button"
        variant="outline-primary"
      >
        
        digdag
      
      </b-button-stub>
    </div>
    
    Date: 
    
    <annotate-buttons-stub
      example="[object Object]"
    />
     
    <b-card-footer-stub
      footertag="div"
    >
      <!---->
       
      <a
        href="https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
        rel="noopener"
        target="_blank"
      >
        blog.trocco.io 
      </a>
       
      <b-button-stub
        class="float-right"
        href="https://twitter.com/intent/tweet?text=👀&hashtags=ml_news&url=https://www.machine-learning.news/example/11111"
        rel="noopener"
        size="sm"
        tag="button"
        target="_blank"
        type="button"
        variant="primary"
      >
        
        Tweet
      
      </b-button-stub>
    </b-card-footer-stub>
  </b-card-stub>
   
  <div
    id="referring-tweet"
  >
    <h2
      class="h4"
    >
      
      Referring Tweets
    
    </h2>
     
    <b-list-group-stub
      tag="div"
    >
      <referring-tweet-stub
        isadmin="true"
        tweet="[object Object]"
      />
    </b-list-group-stub>
  </div>
   
  <div
    id="bookmark-comments"
  >
    <h2
      class="h4"
    >
      
      Bookmark Comments
    
    </h2>
     
    <b-list-group-stub
      tag="div"
    >
      <b-list-group-item-stub
        class="hatena-bookmark-item"
        tag="div"
        target="_self"
      >
        <img
          class="hatena-bookmark-comment-user-icon-img"
          data-src="https://cdn.profile-image.st-hatena.com/users/polamjag/profile.png"
          lazy="loading"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
         
        <div
          class="hatena-bookmark-user-link-and-comment-container"
        >
          <a
            class="hatena-bookmark-user-link"
            href="http://b.hatena.ne.jp/polamjag"
            rel="noopener"
            target="_blank"
          >
            
            id:polamjag
          
          </a>
           
          <span
            class="hatena-bookmark-comment"
          >
            
            “統合”
          
          </span>
           
          <span
            class="hatena-bookmark-timestamp"
          >
            
            2020/08/26 14:29
          
          </span>
        </div>
      </b-list-group-item-stub>
    </b-list-group-stub>
  </div>
   
  <!---->
   
  <b-list-group-stub
    tag="div"
  />
</div>
`;

exports[`ExamplePage#Amplify amplifyを通っていないときはアノテーションボタンを出さない 1`] = `
<div
  class="mx-auto example"
>
  <b-card-stub
    bodytag="div"
    footertag="div"
    headertag="div"
    subtitletag="h6"
    subtitletextvariant="muted"
    tag="article"
    titletag="h4"
  >
    <h1
      class="h4"
    >
      
      Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
    
    </h1>
     
    <!---->
     
    <p
      class="card-text"
    >
      
      Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
    
    </p>
     
    <div>
      <a
        class="hatena-bookmark-link"
        href="https://b.hatena.ne.jp/entry/s/blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
        rel="noopener"
        target="_blank"
      >
        5 users
      </a>
      :
      
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
    </div>
     
    <div>
      <span
        class="tweets-count"
      >
        2 mentions
      </span>
      :
      
      <twitter-icon-stub
        tweet="[object Object]"
      />
      <twitter-icon-stub
        tweet="[object Object]"
      />
    </div>
     
    <div>
      
      Keywords: 
      
      <b-button-stub
        class="keyword"
        href="/search?query=embulk"
        size="sm"
        tag="button"
        target="_self"
        type="button"
        variant="outline-primary"
      >
        
        embulk
      
      </b-button-stub>
      <b-button-stub
        class="keyword"
        href="/search?query=digdag"
        size="sm"
        tag="button"
        target="_self"
        type="button"
        variant="outline-primary"
      >
        
        digdag
      
      </b-button-stub>
    </div>
    
    Date: 
    
    <!---->
     
    <b-card-footer-stub
      footertag="div"
    >
      <!---->
       
      <a
        href="https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
        rel="noopener"
        target="_blank"
      >
        blog.trocco.io 
      </a>
       
      <b-button-stub
        class="float-right"
        href="https://twitter.com/intent/tweet?text=👀&hashtags=ml_news&url=https://www.machine-learning.news/example/11111"
        rel="noopener"
        size="sm"
        tag="button"
        target="_blank"
        type="button"
        variant="primary"
      >
        
        Tweet
      
      </b-button-stub>
    </b-card-footer-stub>
  </b-card-stub>
   
  <div
    id="referring-tweet"
  >
    <h2
      class="h4"
    >
      
      Referring Tweets
    
    </h2>
     
    <b-list-group-stub
      tag="div"
    >
      <referring-tweet-stub
        tweet="[object Object]"
      />
    </b-list-group-stub>
  </div>
   
  <div
    id="bookmark-comments"
  >
    <h2
      class="h4"
    >
      
      Bookmark Comments
    
    </h2>
     
    <b-list-group-stub
      tag="div"
    >
      <b-list-group-item-stub
        class="hatena-bookmark-item"
        tag="div"
        target="_self"
      >
        <img
          class="hatena-bookmark-comment-user-icon-img"
          data-src="https://cdn.profile-image.st-hatena.com/users/polamjag/profile.png"
          lazy="loading"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
         
        <div
          class="hatena-bookmark-user-link-and-comment-container"
        >
          <a
            class="hatena-bookmark-user-link"
            href="http://b.hatena.ne.jp/polamjag"
            rel="noopener"
            target="_blank"
          >
            
            id:polamjag
          
          </a>
           
          <span
            class="hatena-bookmark-comment"
          >
            
            “統合”
          
          </span>
           
          <span
            class="hatena-bookmark-timestamp"
          >
            
            2020/08/26 14:29
          
          </span>
        </div>
      </b-list-group-item-stub>
    </b-list-group-stub>
  </div>
   
  <!---->
   
  <b-list-group-stub
    tag="div"
  />
</div>
`;

exports[`ExamplePage#Amplify amplifyを通っていないときはアノテーションボタンを出さない 2`] = `
<div
  class="mx-auto example"
>
  <b-card-stub
    bodytag="div"
    footertag="div"
    headertag="div"
    subtitletag="h6"
    subtitletextvariant="muted"
    tag="article"
    titletag="h4"
  >
    <h1
      class="h4"
    >
      
      Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
    
    </h1>
     
    <!---->
     
    <p
      class="card-text"
    >
      
      Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
    
    </p>
     
    <div>
      <a
        class="hatena-bookmark-link"
        href="https://b.hatena.ne.jp/entry/s/blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
        rel="noopener"
        target="_blank"
      >
        5 users
      </a>
      :
      
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
    </div>
     
    <div>
      <span
        class="tweets-count"
      >
        2 mentions
      </span>
      :
      
      <twitter-icon-stub
        tweet="[object Object]"
      />
      <twitter-icon-stub
        tweet="[object Object]"
      />
    </div>
     
    <div>
      
      Keywords: 
      
      <b-button-stub
        class="keyword"
        href="/search?query=embulk"
        size="sm"
        tag="button"
        target="_self"
        type="button"
        variant="outline-primary"
      >
        
        embulk
      
      </b-button-stub>
      <b-button-stub
        class="keyword"
        href="/search?query=digdag"
        size="sm"
        tag="button"
        target="_self"
        type="button"
        variant="outline-primary"
      >
        
        digdag
      
      </b-button-stub>
    </div>
    
    Date: 
    
    <annotate-buttons-stub
      example="[object Object]"
    />
     
    <b-card-footer-stub
      footertag="div"
    >
      <!---->
       
      <a
        href="https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
        rel="noopener"
        target="_blank"
      >
        blog.trocco.io 
      </a>
       
      <b-button-stub
        class="float-right"
        href="https://twitter.com/intent/tweet?text=👀&hashtags=ml_news&url=https://www.machine-learning.news/example/11111"
        rel="noopener"
        size="sm"
        tag="button"
        target="_blank"
        type="button"
        variant="primary"
      >
        
        Tweet
      
      </b-button-stub>
    </b-card-footer-stub>
  </b-card-stub>
   
  <div
    id="referring-tweet"
  >
    <h2
      class="h4"
    >
      
      Referring Tweets
    
    </h2>
     
    <b-list-group-stub
      tag="div"
    >
      <referring-tweet-stub
        isadmin="true"
        tweet="[object Object]"
      />
    </b-list-group-stub>
  </div>
   
  <div
    id="bookmark-comments"
  >
    <h2
      class="h4"
    >
      
      Bookmark Comments
    
    </h2>
     
    <b-list-group-stub
      tag="div"
    >
      <b-list-group-item-stub
        class="hatena-bookmark-item"
        tag="div"
        target="_self"
      >
        <img
          class="hatena-bookmark-comment-user-icon-img"
          data-src="https://cdn.profile-image.st-hatena.com/users/polamjag/profile.png"
          lazy="loading"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
         
        <div
          class="hatena-bookmark-user-link-and-comment-container"
        >
          <a
            class="hatena-bookmark-user-link"
            href="http://b.hatena.ne.jp/polamjag"
            rel="noopener"
            target="_blank"
          >
            
            id:polamjag
          
          </a>
           
          <span
            class="hatena-bookmark-comment"
          >
            
            “統合”
          
          </span>
           
          <span
            class="hatena-bookmark-timestamp"
          >
            
            2020/08/26 14:29
          
          </span>
        </div>
      </b-list-group-item-stub>
    </b-list-group-stub>
  </div>
   
  <!---->
   
  <b-list-group-stub
    tag="div"
  />
</div>
`;

exports[`ExamplePage#Amplify amplifyを通っているときはアノテーションボタンを出す 1`] = `
<div
  class="mx-auto example"
>
  <b-card-stub
    bodytag="div"
    footertag="div"
    headertag="div"
    subtitletag="h6"
    subtitletextvariant="muted"
    tag="article"
    titletag="h4"
  >
    <h1
      class="h4"
    >
      
      Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
    
    </h1>
     
    <!---->
     
    <p
      class="card-text"
    >
      
      Embulk & Digdag Online Meetup 2020【イベントレポート】 | trocco(トロッコ)
    
    </p>
     
    <div>
      <a
        class="hatena-bookmark-link"
        href="https://b.hatena.ne.jp/entry/s/blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
        rel="noopener"
        target="_blank"
      >
        5 users
      </a>
      :
      
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
      <hatena-bookmark-icon-stub
        bookmark="[object Object]"
      />
    </div>
     
    <div>
      <span
        class="tweets-count"
      >
        2 mentions
      </span>
      :
      
      <twitter-icon-stub
        tweet="[object Object]"
      />
      <twitter-icon-stub
        tweet="[object Object]"
      />
    </div>
     
    <div>
      
      Keywords: 
      
      <b-button-stub
        class="keyword"
        href="/search?query=embulk"
        size="sm"
        tag="button"
        target="_self"
        type="button"
        variant="outline-primary"
      >
        
        embulk
      
      </b-button-stub>
      <b-button-stub
        class="keyword"
        href="/search?query=digdag"
        size="sm"
        tag="button"
        target="_self"
        type="button"
        variant="outline-primary"
      >
        
        digdag
      
      </b-button-stub>
    </div>
    
    Date: 
    
    <annotate-buttons-stub
      example="[object Object]"
    />
     
    <b-card-footer-stub
      footertag="div"
    >
      <!---->
       
      <a
        href="https://blog.trocco.io/dev-articles/embulk-digdag-meetup-2020"
        rel="noopener"
        target="_blank"
      >
        blog.trocco.io 
      </a>
       
      <b-button-stub
        class="float-right"
        href="https://twitter.com/intent/tweet?text=👀&hashtags=ml_news&url=https://www.machine-learning.news/example/11111"
        rel="noopener"
        size="sm"
        tag="button"
        target="_blank"
        type="button"
        variant="primary"
      >
        
        Tweet
      
      </b-button-stub>
    </b-card-footer-stub>
  </b-card-stub>
   
  <div
    id="referring-tweet"
  >
    <h2
      class="h4"
    >
      
      Referring Tweets
    
    </h2>
     
    <b-list-group-stub
      tag="div"
    >
      <referring-tweet-stub
        isadmin="true"
        tweet="[object Object]"
      />
    </b-list-group-stub>
  </div>
   
  <div
    id="bookmark-comments"
  >
    <h2
      class="h4"
    >
      
      Bookmark Comments
    
    </h2>
     
    <b-list-group-stub
      tag="div"
    >
      <b-list-group-item-stub
        class="hatena-bookmark-item"
        tag="div"
        target="_self"
      >
        <img
          class="hatena-bookmark-comment-user-icon-img"
          data-src="https://cdn.profile-image.st-hatena.com/users/polamjag/profile.png"
          lazy="loading"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
         
        <div
          class="hatena-bookmark-user-link-and-comment-container"
        >
          <a
            class="hatena-bookmark-user-link"
            href="http://b.hatena.ne.jp/polamjag"
            rel="noopener"
            target="_blank"
          >
            
            id:polamjag
          
          </a>
           
          <span
            class="hatena-bookmark-comment"
          >
            
            “統合”
          
          </span>
           
          <span
            class="hatena-bookmark-timestamp"
          >
            
            2020/08/26 14:29
          
          </span>
        </div>
      </b-list-group-item-stub>
    </b-list-group-stub>
  </div>
   
  <!---->
   
  <b-list-group-stub
    tag="div"
  />
</div>
`;
